src/data/constants.js

```diff
@@ -0,0 +1,64 @@
+import { getConfig } from '@edx/frontend-platform';
+
+export const API_BASE_URL = getConfig().LMS_BASE_URL;
+
+export const LoadingStatus = {
+  LOADING: 'loading',
+  LOADED: 'loaded',
+  FAILED: 'failed',
+  DENIED: 'denied',
+};
+
+export const ThreadOrdering = {
+  BY_LAST_ACTIVITY: 'sort_by_last_activity',
+  BY_COMMENT_COUNT: 'sort_by_comment_count',
+  BY_VOTE_COUNT: 'sort_by_vote_count',
+};
+
+export const ThreadView = {
+  UNREAD: 'unread',
+  UNANSWERED: 'unanswered',
+};
+
+export const MyThreadsFilter = {
```

Comment:
Any reason why we're changing from `Posts` (`All posts`, `My posts`) to `Threads` in this PR?
Can you adjust it to match the mockups?

Reply:
I changed it because the terminology used in the mockups do not match the terminology used in the current edx-platfor API, and in general, the definitions of the words in the English language.

The [edx-platform API](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/discussion/rest_api/views.py#L145) uses the term 'Thread' to describe what the mockups label as a 'Post'.

The definition of the term 'Post' on [Cambridge Dictionary](https://dictionary.cambridge.org/dictionary/english/post) is:
```
something such as a message or picture that you publish on a website or using social media
```

Conversely, the definition of 'Thread' on [Cambridge Dictionary](https://dictionary.cambridge.org/dictionary/english/thread) is:
```
The thread of a book, discussion, speech, etc. is its story or the way that it develops, one part connecting with another
```

This can easily be changed back, but I believe this terminology of using 'Thread' as a collection of messages from users, the first of which is a 'Post', and subsequent ones being either a 'Comment' or a 'Reply' is better. Please advise if this needs to be changed.

Reply:
I still think we should follow the mockups, but this is something I need more opinions on. @xitij2000 can you pitch in here?
Also, can you ask this upstream somewhere? Make sure to ping me and Kshitij there.

Reply:
> I changed it because the terminology used in the mockups do not match the terminology used in the current edx-platfor API, and in general, the definitions of the words in the English language.

We definitely need to follow the mockups. The terms used in the API are mostly inconsequential since:
- End-users will not see the API
- We are doing a redesign of the UI and potentially the APIs as well

> The [edx-platform API](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/discussion/rest_api/views.py#L145) uses the term 'Thread' to describe what the mockups label as a 'Post'.

Yes, the API has topics, threads and comments. The UI doesn't use the same terms, we're following the mockup, not the existing API. What you should look at for reference is the existing UI, which uses the same terms.

This is the topics UI:
![Screenshot_2020-09-02 Discussion DemoX OpenCraft - Demo Instance](https://user-images.githubusercontent.com/118837/91935035-4acb0200-ecdc-11ea-97c8-fb993370c18d.png)

This is the posts UI. Note that the term here is "show all posts":
![Screenshot_2020-09-02 Discussion DemoX OpenCraft - Demo Instance(1)](https://user-images.githubusercontent.com/118837/91935130-81a11800-ecdc-11ea-9d67-7c7c1f8cbd67.png)

> This can easily be changed back, but I believe this terminology of using 'Thread' as a collection of messages from users, the first of which is a 'Post', and subsequent ones being either a 'Comment' or a 'Reply' is better. Please advise if this needs to be changed.

I understand, but that discussion is not in scope here. The existing UI and new UI are consistent in how they name things and we need to maintain that.